packages/@core.keystone/errors.js

```diff
@@ -0,0 +1,136 @@
+ *         }
+ *     ],
+ *     mutations: [
+ *         {
+ *             access: true,
+ *             schema: 'myCustomAction(data: MyCustomActionInput!): User',
+ *             doc: {
+ *                 summary: 'Does this and that, shortly',
+ *                 description: 'More detailed explanation goes here',
+ *                 // Pass declared errors object here to get formatted documentation in GraphiQL
+ *                 errors,
+ *             },
+ *             resolver: async (parent, args, context) => {
+ *                 // validate passed arguments
+ *                 if (!normalizePhone(args.phone)) {
+ *                     // Use our customized error class and pass appropriate error into it
+ *                     throw new GQLError(errors.WRONG_PHONE_FORMAT)
+ *                 }
+ *
+ *                 // find some record
+ *                 const user = await User.getOne(context, { id: args.userId })
+ *                 if (!user) {
+ *                     throw new GQLError(errors.UNABLE_TO_FIND_USER)
+ *                 }
+ *
+ *                 // execute some another actions, that is not guaranteed to be succeed
+ *                 try {
+ *                     executeSomeAnotherAction
+ *                 } catch(e) {
+ *                     throw new GQLError({
+ *                         ...errors.UNABLE_EXECUTE_SOME_PROCEDURE,
+ *                         internalError: e
+ *                     })
+ *                 }
+ *             }
+ *         }
+ *     ]
+ * })
+ * ```
+ */
+const { ApolloError } = require('apollo-server-errors')
+const { extractReqLocale } = require('@condo/domains/common/utils/locale')
+const conf = require('@core/config')
+const { getTranslations } = require('@condo/domains/common/utils/localesLoader')
+const template = require('lodash/template')
+const templateSettings = require('lodash/templateSettings')
+
+// Matches placeholder `{name}` in string, we are going to interpolate
+templateSettings.interpolate = /{([\s\S]+?)}/g
+
+// Unable to find a record, whose identifier is specified in some argument of query or mutation
+const NOT_FOUND = 'NOT_FOUND'
+// User input cannot be processed by server by following reasons:
+// wrong format, not enough data, conflicts with data storage constraints (duplicates etc)
+const BAD_USER_INPUT = 'BAD_USER_INPUT'
+// Generic error, that something went wrong at server side, though user input was correct
+const INTERNAL_ERROR = 'INTERNAL_ERROR'
+// Access denied
+const FORBIDDEN = 'FORBIDDEN'
+
+
+/**
+ * First level of error classification, used in custom GraphQL queries or mutations
+ * Second level of classification will be specific to domain in question
+ * Only generic error kinds are listed
+ * Conceptually, it conforms to HTTP standard for error codes
+ * @readonly
+ * @enum {String}
+ */
+const GQLErrorCode = {
+    NOT_FOUND,
```

Comment:
I don't understand how to unambiguously match the error to the NOT_FOUND code. For example, a user passes a link to an object to the API, but he does not have access to it. For example, this is the creation of a ticket, and he passes the ID of the organization to which he does not have access. 

I think it's a BAD_USER_INPUT. Because it's a hacker who tries to check the API. 

To be honest, I do not know why we might need this code. And I would introduce it if we find a case when we use it.

Reply:
And the default Apollo codes: https://www.apollographql.com/docs/apollo-server/data/errors/#error-codes doesn't contain NOT_FOUND code

Reply:
Yes, they don't have it, because GraphQL returns either empty arrays or null.
I will get the `NOT_FOUND` out of set.

Reply:
But, in some mutations we returning errors about something is not found. Will see, whether we need `NOT_FOUND` error code during transition of all custom mutations and queries from all domains.

Reply:
I think, it will be a type `NOT_FOUND` (2-nd level classification), not the code.